Automattic/wp-calypso
# Editor: Move the HTML Toolbar methods into their own utility component

The Post Editor is not the only editor that uses the HTML Toolbar (see #21663).

This PR moves most of the reusable methods out of `EditorHtmlToolbar` and into an utility folder that includes:
- The `utils.js` file, with all the reusable methods.
- Two reusable Dialog components. 

The `components/html-toolbar` folder does not contain any `index.jsx` file, or, in other words, there is no generic `HtmlToolbar`.
The reason is that the Editor uses a LOT of custom stuff: the Insert menu, the sticky/scroll-y behaviour, etc.
I could have done a more complicated refactor, trying to make the entire toolbar 100% reusable, but I'm not sure it would make sense with Gutenberg around the corner.
On the other hand, it seems weird to just dumbly duplicate half of the Post Editor toolbar into the Comment Editor toolbar.

## Testing instructions

This shouldn't change the behaviour of the Editor in any way, so the idea is to do a smoke test on the Editor in HTML mode, and see if all the buttons still work, and work as expected.
